packages/twenty-front/src/modules/object-record/record-index/options/hooks/useRecordIndexOptionsForBoard.ts

```diff
@@ -0,0 +1,156 @@
+import { useCallback, useMemo } from 'react';
+import { OnDragEndResponder } from '@hello-pangea/dnd';
+import { useRecoilState } from 'recoil';
+
+import { mapBoardFieldDefinitionsToViewFields } from '@/companies/utils/mapBoardFieldDefinitionsToViewFields';
+import { useColumnDefinitionsFromFieldMetadata } from '@/object-metadata/hooks/useColumnDefinitionsFromFieldMetadata';
+import { useObjectMetadataItemOnly } from '@/object-metadata/hooks/useObjectMetadataItemOnly';
+import { FieldMetadata } from '@/object-record/record-field/types/FieldMetadata';
+import { recordIndexFieldDefinitionsState } from '@/object-record/record-index/states/recordIndexFieldDefinitionsState';
+import { ColumnDefinition } from '@/object-record/record-table/types/ColumnDefinition';
+import { useViewFields } from '@/views/hooks/internal/useViewFields';
+
+type useRecordIndexOptionsForBoardParams = {
+  objectNameSingular: string;
+  viewBarId: string;
+};
+
+export const useRecordIndexOptionsForBoard = ({
```

Comment:
I'm keeping the "ForBoard" suffix for now but the idea is to share this between table and board.
This refactoring is too heavy for now.